nightly-build.yml

```diff
@@ -0,0 +1,145 @@
+schedules:
+  - cron: "0 0 * * *"
+    displayName: Daily Midnight AzCopy Build
+    branches:
+      include:
+        - dev
+    always: true
+
+jobs:
+  - job: Nightly
+    # allow maximum build time, in case we have build congestion
+    timeoutInMinutes: 360
+    # Creating strategies for GOOS: Windows Server 2019 /macOS X Mojave 10.14/Ubuntu 20.04
```

Comment:
can we use the latest version of MacOS instead? Big Sur?

Reply:
Believe it or not but this is the latest which Azure supports :P